_python/main/views.py

```diff
@@ -255,25 +257,118 @@ def render_with_actions(request, template_name, context=None, content_type=None,
 
 ### views ###
 
-@perms_test(
-    {'args': ['resource.id'], 'results': {200: ['resource.casebook.owner', 'other_user', 'admin_user', None]}},
-    # only editor can get annotations for draft resource
-    {'args': ['full_casebook_with_draft.drafts.resources.first.id'], 'results': {200: ['full_casebook_with_draft.drafts.resources.first.casebook.owner', 'admin_user'], 403: ['other_user'], 'login': [None]}},
-)
-@api_view(['GET'])
-def annotations(request, resource_id, format=None):
-    """
-        /resources/:resource_id/annotations view.
-        Was: app/controllers/content/annotations_controller.rb
-    """
-    resource = get_object_or_404(Resource.objects.select_related('casebook'), pk=resource_id)
 
-    # check permissions
-    if not resource.viewable_by(request.user):
-        return login_required_response(request)
+class AnnotationListView(APIView):
+
+    @method_decorator(perms_test(
+        {'args': ['resource'], 'results': {200: ['resource.casebook.owner', 'other_user', 'admin_user', None]}},
+        {'args': ['full_casebook_with_draft.drafts.resources.first'], 'results': {200: ['full_casebook_with_draft.drafts.resources.first.casebook.owner', 'admin_user'], 403: ['other_user'], 'login': [None]}},
```

Comment:
Nit: the user here could be full_casebook_with_draft.drafts.owner

Reply:
👍